mono/metadata/domain.c

```diff
@@ -382,18 +382,7 @@ mono_domain_create (void)
 	}
 	mono_appdomains_unlock ();
 
-#ifdef HAVE_BOEHM_GC
-	/*
-	 * Boehm doesn't like roots inside GC allocated objects, and alloc_fixed returns
-	 * a GC_MALLOC-ed object, contrary to the api docs. This causes random crashes when
-	 * running the corlib test suite.
-	 * To solve this, we pass a NULL descriptor, and don't register roots.
-	 */
-	domain = (MonoDomain *)mono_gc_alloc_fixed (sizeof (MonoDomain), NULL, MONO_ROOT_SOURCE_DOMAIN, "domain object");
-#else
 	domain = (MonoDomain *)mono_gc_alloc_fixed (sizeof (MonoDomain), domain_gc_desc, MONO_ROOT_SOURCE_DOMAIN, "domain object");
-	mono_gc_register_root ((char*)&(domain->MONO_DOMAIN_FIRST_GC_TRACKED), G_STRUCT_OFFSET (MonoDomain, MONO_DOMAIN_LAST_GC_TRACKED) - G_STRUCT_OFFSET (MonoDomain, MONO_DOMAIN_FIRST_GC_TRACKED), MONO_GC_DESCRIPTOR_NULL, MONO_ROOT_SOURCE_DOMAIN, "misc domain fields");
```

Comment:
Actually, it looks like everything but one value between `MONO_DOMAIN_FIRST_GC_TRACKED` and `MONO_DOMAIN_FIRST_GC_TRACKED` is a `MonoGHashTable`. The remaining item is the root `static_data_array` for domain statics which is allocated with `mono_gc_alloc_fixed`.

I can remove this change for now (and open another PR that cleans it up further), but don't think any of these items need scanned by GC as all should be immovable memory already tracked by the GC.